### PR TITLE
feat(typescript-sdk): implement subscrtions manager interface and websocket implementation

### PR DESCRIPTION
**Pull request by fbozic**

### CODE REVIEW

File: examples/only-peers-simple/README.md [hunk 1]
```diff
@@ -1,56 +1,47 @@
...
 To build the production-ready version of the application, run:
 
 ```
-yarn build
+pnmp build
 ```
 This will create a dist directory containing the compiled and optimized version of the application. To preview the production build, you can run:
 
-```
+pnmp
```
Comment (MatejVukosav) [inline examples/only-peers-simple/README.md]: wrong edit?

File: crates/primitives/src/events.rs [hunk 2]
```diff
@@ -17,7 +17,7 @@ pub struct ApplicationEventPayload {
 }
 
 #[derive(Clone, Serialize, Deserialize, Debug)]
-#[serde(rename_all = "camelCase")]
+#[serde(tag = "type", content = "payload", rename_all = "snake_case")]
```
Comment (miraclx) [inline crates/primitives/src/events.rs]: use `#[serde(untagged)]` on `NodeEvent`

so notifs that include `application_id` are automatically inferred as `ApplicationEvent`

Also just pointing out that the pattern across the codebase is `type` & `data`, to be sure you've considered that.

Also, use `PascalCase` when serializing a type in value position.

File: examples/only-peers-simple/README.md (hunk 1 above)
Comment (fbozic) [reply examples/only-peers-simple/README.md]: yes

File: crates/primitives/src/events.rs (hunk 2 above)
Comment (fbozic) [reply crates/primitives/src/events.rs]: I'll update it here, and I've added comment to this issue to follow up: https://github.com/calimero-network/core/issues/115#issuecomment-2049478257

**Pull request merged**